src/field/field2d.cxx

```diff
@@ -249,6 +249,7 @@ Field2D &Field2D::operator=(const Field2D &rhs) {
 
   // Copy the data and data sizes
   fieldmesh = rhs.fieldmesh;
+  fielddatamesh = rhs.fielddatamesh;
```

Comment:
Do we need to check that fieldmesh/fielddatamesh are identical? If so why do we need two variables?

Reply:
They definitely should be the same. We should `ASSERT1` in `Field2D` and `Field3D`.

There are two because there's one in each of the two base classes. I think this points towards our class hierarchy being a little weird

Reply:
I was wondering if this might be an opportunity to address the class structure? We're currently introducing more opportunities to form an inconsistent state (albeit with appropriate checks to try to guard against this). All vectors and fields apart from fieldperp have fielddata and all fields have Field. I'd guess FieldPerp should also have fielddata (and if not why not?) and if so then it looks like `FieldData` is the generic stuff to all field like objects (which would presumably include getting the single relevant mesh) whilst `Field` provides the things fields have that vectors don't (coordinates, anything else?).

Reply:
Or could Field inherit from FieldData?